bluebottle/members/tests/test_api.py

```diff
@@ -291,6 +314,42 @@ def test_create_twice(self):
             user_id
         )
 
+    def test_create_twice_different_case(self):
+        email = 'test@example.com'
+        password = 'test@example.com'
+
+        response = self.client.post(
+            reverse('user-user-create'),
+            {'email': email, 'password': password, 'email_confirmation': email}
+        )
+        user_id = str(response.json()['id'])
+
+        response = self.client.post(
+            reverse('user-user-create'),
+            {
+                'email': email.replace('test', 'Test'),
+                'password': password,
+                'email_confirmation': email.replace('test', 'Test')
+            }
+        )
+
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+        self.assertEqual(
+            response.json()['non_field_errors'][0]['email'],
+            email
+        )
+
+        self.assertEqual(
+            response.json()['non_field_errors'][0]['type'],
+            'email'
+        )
+
+        self.assertEqual(
+            response.json()['non_field_errors'][0]['id'],
+            user_id
+        )
+
```

Comment:
Maybe another test to specifically test the flow where user uses email sign-up first and then Facebook or v.v.? I think that is most common scenario. Or will that be using the same UserCreateSerializer?

Reply:
Facebook login is case-insensitive already. But is really hard to test from our code. There are tests in social-django though that verify this.